samwise.js

```diff
@@ -5,7 +5,8 @@
  * Released under the MIT license.
```

Comment:
SCRIPT-I'm a bit concerned about the potential performance impact of this loop. Have you thought about optimizing it?